src/web/components/elements/MultiImageBlockComponent.tsx

```diff
@@ -0,0 +1,294 @@
+    // designType,
+    caption,
+    images,
+    pillar,
+}: Props) => {
+    const imageCount = images.length;
+
+    switch (imageCount) {
+        case 1:
+            return (
+                <div
+                    className={css`
+                        img {
+                            object-fit: cover;
+                            width: 100%;
+                        }
+                    `}
+                >
+                    <ImageComponent
+                        display="standard"
+                        element={images[0]}
+                        pillar={pillar}
+                        hideCaption={true}
+                        role={images[0].role}
+                    />
+                    {caption && (
+                        <Caption
+                            display="standard"
+                            captionText={caption}
+                            pillar={pillar}
+                            shouldLimitWidth={false}
+                        />
+                    )}
+                </div>
+            );
+        case 2:
+            return (
+                <div
+                    className={css`
+                        img {
+                            object-fit: cover;
+                            width: 100%;
+                        }
+                    `}
+                >
+                    <SideBySideGrid>
+                        <GridItem area="first">
+                            <ImageComponent
+                                display="standard"
+                                element={images[0]}
+                                pillar={pillar}
+                                hideCaption={true}
+                                role={images[0].role}
+                            />
+                        </GridItem>
+                        <GridItem area="second">
+                            <ImageComponent
+                                display="standard"
+                                element={images[1]}
+                                pillar={pillar}
+                                hideCaption={true}
+                                role={images[1].role}
+                            />
+                        </GridItem>
+                    </SideBySideGrid>
+                    {caption && (
+                        <Caption
+                            display="standard"
+                            captionText={caption}
+                            pillar={pillar}
+                            shouldLimitWidth={false}
+                        />
+                    )}
+                </div>
+            );
+        case 3:
+            return (
+                <div
+                    className={css`
+                        img {
+                            object-fit: cover;
+                            width: 100%;
+                        }
+                    `}
+                >
+                    <OneAboveTwoGrid>
```

Comment:
What about scenarios where it's "two above one"? Example near the bottom of [this article](https://www.theguardian.com/travel/2019/jan/25/mountain-rescue-why-bothies-need-a-helping-hand-a-photo-essay).

Reply:
Dang. That example really messes up my api!

Either:
1. You just accept that when there are 3 images they will always appear 1 above 2
2. You refactor the api here to have a way to communicate the fact you want 2 above 1 or vice versa
3. The journo constructs the look they want using a combination of 1 image + 2

Actually, 1 & 3 are complementary.

Reply:
Perhaps it's possible to parse this using the image `role`? In the "two above one" example the first two are `role: halfWidth`, and the third has no `role` which means full width.

> The journo constructs the look they want using a combination of 1 image + 2

Would this mean losing the shared caption?

Reply:
> Would this mean losing the shared caption?

It would mean you'd lose some data integrity because you'd have to deliberately not set a caption for the set of 2 and then have the caption for the 1 apply to all of them. Which is basically where we're at right now.

I suppose you could enhance the api to have a `twoAtTop` boolean. It would make cleaning the data harder but is probably the best (most flexible) way to present the new element in Composer

Reply:
Is there a way to make the container layout-agnostic?

```tsx
type Width = Half | Full;

interface Image {
  width: Width;
  sources: ...;
  data: ...;
}

interface Props {
  images: Image[];
  caption: string | null;
}

const halfStyles = css`
  flex-basis: 50%;
`;

const fullStyles = css`
  flex-basis: 100%;
`;

const styles = css`
  display: flex;
  flex-wrap: wrap;
`;

const imgStyles = (width: Width) => {
  switch (width) {
    case Half:
      return halfStyles;
    case Full:
    default:
      return fullStyles;
  }
}

const MultiImage = (props: Props) =>
  <figure css={styles}>
    {images.map(image => <img css={imgStyles(image.width)} {...image.data} />}
    {props.caption ? <figcaption>{props.caption}</figcaption> : null}
  </figure>

```

I have a vague memory of one of these having _five_ images in it...

Reply:
Basically, we're going to have to support any combination of images in half width or non-half-width combinations...

https://github.com/guardian/frontend/blob/3095a28e951b63ae993aa3d3fcc59a3cf83afc4b/static/src/stylesheets/inline/article-photo-essay-garnett.scss#L222

https://github.com/guardian/frontend/blob/08b3732d113c28c5165ab2ce8e9b7b52909da336/common/app/views/support/HtmlCleaner.scala#L467

If they wanted 4 half width image, a big image in the middle then 4 more half width they could right now. So we either descope this or we have to adjust the API as @JamieB-gu mentions to handle the combo...

Probably latter as we've got existing content to support and we don't wanna have to split the components.